
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>The Metagenome Assembly Workflow &#8212; NMDC Workflows 0.1 documentation</title>
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Metagenome Annotation Workflow" href="4_MetaGAnnotation_index.html" />
    <link rel="prev" title="Metagenome assembled genomes generation workflow" href="2_MAG_index.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="the-metagenome-assembly-workflow">
<h1>The Metagenome Assembly Workflow<a class="headerlink" href="#the-metagenome-assembly-workflow" title="Permalink to this headline">¶</a></h1>
<div class="section" id="summary">
<h2>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">¶</a></h2>
<p>This workflow is developed by Brian Foster at JGI and original from his <a class="reference external" href="https://gitlab.com/bfoster1/wf_templates/tree/master/templates">repo</a>. It take paired-end reads runs error correction by bbcms (BBTools). The clean reads are assembled by MetaSpades. After assembly, the reads are mapped back to contigs by bbmap (BBTools) for coverage information.</p>
</div>
<div class="section" id="workflow-diagram">
<h2>Workflow Diagram<a class="headerlink" href="#workflow-diagram" title="Permalink to this headline">¶</a></h2>
<a class="reference internal image-reference" href="../_images/3_MetaGAssemly_workflow_assembly.png"><img alt="Metagenome assembly workflow dependencies" src="../_images/3_MetaGAssemly_workflow_assembly.png" style="width: 914.4px; height: 266.4px;" /></a>
</div>
<div class="section" id="workflow-dependencies">
<h2>Workflow Dependencies<a class="headerlink" href="#workflow-dependencies" title="Permalink to this headline">¶</a></h2>
<div class="section" id="third-party-software">
<h3>Third party software<a class="headerlink" href="#third-party-software" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>MetaSPades v3.13.0 (GPLv2)</p></li>
<li><p>BBTools:38.44 <a class="reference external" href="https://bitbucket.org/berkeleylab/jgi-bbtools/src/master/license.txt">(BSD-3-Clause-LBNL)</a></p></li>
</ul>
</div>
<div class="section" id="database">
<h3>Database<a class="headerlink" href="#database" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>None</p></li>
</ul>
</div>
</div>
<div class="section" id="workflow-availability">
<h2>Workflow Availability<a class="headerlink" href="#workflow-availability" title="Permalink to this headline">¶</a></h2>
<p>The workflow is available in GitHub:
<a class="reference external" href="https://github.com/microbiomedata/metaAssembly">https://github.com/microbiomedata/metaAssembly</a></p>
<p>The container is available at Docker Hub (microbiomedata):</p>
<p><a class="reference external" href="https://hub.docker.com/r/microbiomedata/spades">https://hub.docker.com/r/microbiomedata/spades</a></p>
<p><a class="reference external" href="https://hub.docker.com/r/microbiomedata/bbtools">https://hub.docker.com/r/microbiomedata/bbtools</a></p>
</div>
<div class="section" id="test-datasets">
<h2>Test datasets<a class="headerlink" href="#test-datasets" title="Permalink to this headline">¶</a></h2>
<p>Zymobiomics mock-community DNA control <a class="reference external" href="https://www.ebi.ac.uk/ena/browser/view/SRR7877884">(SRR7877884)</a></p>
</div>
<div class="section" id="details">
<h2>Details<a class="headerlink" href="#details" title="Permalink to this headline">¶</a></h2>
<p>The .wdl (Workflow Description Language) file includes five tasks, <cite>bbcms</cite>, <cite>assy</cite>, <cite>create_agp</cite>, <cite>read_mapping_pairs</cite>, and <cite>make_output</cite>.</p>
<p>The <cite>bbcms</cite> tasks takes interleaved fastq inputs and perform error correction, reformat interleaved fastq into two output fastq files for paired-end reqads for next tasks.
The <cite>assy</cite> task performs metaSPAdes assembly.
Contigs and Scaffolds output of metaSPAdes are consumed by <cite>create_agp</cite> task to rename the fasta header and generate <a class="reference external" href="https://www.ncbi.nlm.nih.gov/assembly/agp/AGP_Specification/">AGP</a> format which describes the assembly.
The <cite>read_mapping_pairs</cite> task maps reads back to the final assembly and generated coverage information.
The final <cite>make_output</cite> task will pop all above output files into specified directory.</p>
<div class="section" id="inputs">
<h3>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h3>
<div class="highlight-JSON notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="nt">&quot;jgi_meta.input_file&quot;</span><span class="p">:[</span><span class="s2">&quot;/path/to/SRR7877884.fastq.gz&quot;</span><span class="p">],</span>
  <span class="nt">&quot;jgi_meta.rename_contig_prefix&quot;</span><span class="p">:</span><span class="s2">&quot;SRR7877884&quot;</span><span class="p">,</span>
  <span class="nt">&quot;jgi_meta.outdir&quot;</span><span class="p">:</span><span class="s2">&quot;/path/to/SRR7877884_assembly&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<ul>
<li><p>The json file includes three parts:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>fastq (illumina paired-end interleaved fastq)</p></li>
<li><p>contig prefix for fasta header</p></li>
<li><p>output path</p></li>
</ol>
</div></blockquote>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If the input is paired-end fastq file, it should be in interleaved format. Below is command use the SRR7877884 as example to convert the paired-end reads in two files into one interleaved format fastq file.</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>paste &lt;<span class="o">(</span>zcat SRR7877884_1.fastq.gz <span class="p">|</span> paste - - - -<span class="o">)</span> &lt;<span class="o">(</span>zcat SRR7877884_2.fastq.gz <span class="p">|</span> paste - - - -<span class="o">)</span> <span class="p">|</span> tr <span class="s1">&#39;\t&#39;</span> <span class="s1">&#39;\n&#39;</span> <span class="p">|</span> gzip -c &gt; SRR7877884-int.fastq.gz
</pre></div>
</div>
</div>
<div class="section" id="outputs">
<h3>Outputs<a class="headerlink" href="#outputs" title="Permalink to this headline">¶</a></h3>
<p>The output will have four sub-directories, bbcms, final_assembly, mapping and spades3. The main assembly contigs output is in final_assembly/assembly.contigs.fasta.</p>
<p>Below is a part list of all output files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>├── bbcms
│   ├── berkeleylab-jgi-meta-60ade422cd4e
│   ├── counts.metadata.json
│   ├── input.corr.fastq.gz
│   ├── input.corr.left.fastq.gz
│   ├── input.corr.right.fastq.gz
│   ├── readlen.txt
│   └── unique31mer.txt
├── final_assembly
│   ├── assembly.agp
│   ├── assembly.contigs.fasta
│   ├── assembly.scaffolds.fasta
│   └── assembly.scaffolds.legend
├── mapping
│   ├── covstats.txt
│   ├── pairedMapped.bam
│   ├── pairedMapped.sam.gz
│   ├── pairedMapped_sorted.bam
│   └── pairedMapped_sorted.bam.bai
└── spades3
        ├── assembly_graph.fastg
        ├── assembly_graph_with_scaffolds.gfa
        ├── contigs.fasta
        ├── contigs.paths
        ├── scaffolds.fasta
        └── scaffolds.paths
</pre></div>
</div>
</div>
</div>
<div class="section" id="requirements-for-execution">
<h2>Requirements for Execution<a class="headerlink" href="#requirements-for-execution" title="Permalink to this headline">¶</a></h2>
<ul>
<li><p>Docker or other Container Runtime</p></li>
<li><p>Cromwell or other WDL-capable Workflow Execution Tool</p></li>
<li><p>The memory requirement depends on the input complexity. Here is a simple estimation equation based on kmers.</p>
<p>predicted_mem = (kmers * 2.962e-08 + 1.630e+01) * 1.1 (in GB)</p>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><cite>kmers</cite> variable can get from <cite>kmercountmulti.sh</cite> script from BBTools.</p>
<p>kmercountmulti.sh -k=31 in=your.read.fq.gz</p>
</div>
</div>
<div class="section" id="running-workflow-in-cromwell-on-cori">
<h2>Running Workflow in Cromwell on Cori<a class="headerlink" href="#running-workflow-in-cromwell-on-cori" title="Permalink to this headline">¶</a></h2>
<p>We provide three ways to run the workflow.</p>
<ol class="arabic simple">
<li><p>CromwellJtmShifter/: The Cromwell run in head node send tasks to jtm-task-managers which will manages the tasks running on a computer node and using Shifter to run applications.</p></li>
<li><p>SlurmCromwellShifter/: The submit script will request a node and launch the Cromwell. The Cromwell manages the workflow by using Shifter to run applications.</p></li>
<li><p>CromwellSlurmShifter/: The Cromwell run in head node and manages the workflow by submitting each step of workflow to compute node where applications were ran by Shifter.</p></li>
</ol>
<p>Description of the files in each sub-directory:</p>
<ul class="simple">
<li><p>.wdl file: the WDL file for workflow definition</p></li>
<li><p>.json file: the example input for the workflow</p></li>
<li><p>.conf file: the conf file for running Cromwell.</p></li>
<li><p>.sh file: the shell script for running the example workflow</p></li>
</ul>
</div>
<div class="section" id="version-history">
<h2>Version History<a class="headerlink" href="#version-history" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>1.0.0</p></li>
</ul>
</div>
<div class="section" id="point-of-contact">
<h2>Point of contact<a class="headerlink" href="#point-of-contact" title="Permalink to this headline">¶</a></h2>
<p>Original author: Brian Foster &lt;<a class="reference external" href="mailto:bfoster&#37;&#52;&#48;lbl&#46;gov">bfoster<span>&#64;</span>lbl<span>&#46;</span>gov</a>&gt;</p>
<p>Package maintainer: Chienchi Lo &lt;<a class="reference external" href="mailto:chienchi&#37;&#52;&#48;lanl&#46;gov">chienchi<span>&#64;</span>lanl<span>&#46;</span>gov</a>&gt;</p>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">NMDC Workflows</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="1_RQC_index.html">The Reads QC Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_MAG_index.html">Metagenome assembled genomes generation workflow</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">The Metagenome Assembly Workflow</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#summary">Summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#workflow-diagram">Workflow Diagram</a></li>
<li class="toctree-l2"><a class="reference internal" href="#workflow-dependencies">Workflow Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="#workflow-availability">Workflow Availability</a></li>
<li class="toctree-l2"><a class="reference internal" href="#test-datasets">Test datasets</a></li>
<li class="toctree-l2"><a class="reference internal" href="#details">Details</a></li>
<li class="toctree-l2"><a class="reference internal" href="#requirements-for-execution">Requirements for Execution</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-workflow-in-cromwell-on-cori">Running Workflow in Cromwell on Cori</a></li>
<li class="toctree-l2"><a class="reference internal" href="#version-history">Version History</a></li>
<li class="toctree-l2"><a class="reference internal" href="#point-of-contact">Point of contact</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="4_MetaGAnnotation_index.html">Metagenome Annotation Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="5_Metaproteomics_index.html">Metaproteomic Workflow</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="2_MAG_index.html" title="previous chapter">Metagenome assembled genomes generation workflow</a></li>
      <li>Next: <a href="4_MetaGAnnotation_index.html" title="next chapter">Metagenome Annotation Workflow</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, National Microbiome Data Collaborative.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.1.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../_sources/chapters/3_MetaGAssemly_index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>